sdk/daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/Ast.scala

```diff
@@ -1226,6 +1226,29 @@ object Ast {
       upgradedPackageId: Option[PackageId],
   )
 
+  final case class PkgIdWithMeta(
+      pkgId: Ref.PackageId,
+      metadata: Option[(PackageName, PackageVersion)],
```

Comment:
Why not  
```suggestion
      metadata: Option[PackageMetadata],
```
? 

Why it is optional ?

Reply:
could we move outside the Ast object, as it is not part of the Ast ? 
for example in `com.digitalasset.daml.language.Util` or `com.digitalasset.daml.language`

Reply:
I've moved it out of the Ast object into Util, and I've made metadata no longer optional - it was necessary because a certain code path had an Optional package, but I've fixed that code path now.

I've renamed it to PkgIdWithNameAndVersion - I don't need any other fields in the metadata, so I preferred to pull them out so that it's clear what this case class is for.